### PR TITLE
Add multiargs build e2e tests

### PR DESCRIPTION
**Pull request by ulyssessouza**

### CODE REVIEW

File: pkg/compose/build_classic.go [hunk 1]
```diff
@@ -243,7 +243,7 @@ func imageBuildOptions(options buildx.Options) dockertypes.ImageBuildOptions {
 func toMapStringStringPtr(source map[string]string) map[string]*string {
 	dest := make(map[string]*string)
 	for k, v := range source {
-		v := v
+		// v := v
```
[the remark this answers is not in the document]
Comment (mat007) [reply pkg/compose/build_classic.go:246]: Without that line, what it does is assign the pointer to the last value in `source` to all values in `dest`.
Given that the test has two args with the same value, it explains why it doesn’t fail.

[the remark this answers is not in the document]
Comment (mat007) [reply pkg/compose/build_classic.go:246]: The space for `v` is allocated once, and then each iteration of the loop overwrites `v` with the new value.
But it’s always the same pointer value, as it points to the same memory location.
Therefore every time we assign the pointer, it’s actually the same pointer. And after the loop is finished, the variable pointed to by that pointer holds the value of the last element in the range.

File: pkg/e2e/compose_build_test.go [hunk 2]
```diff
@@ -66,6 +66,21 @@ func TestLocalComposeBuild(t *testing.T) {
 		res.Assert(t, icmd.Expected{Out: `"FOO": "BAR"`})
 	})
 
+	t.Run("build with multiple build-args ", func(t *testing.T) {
+		for i := 0; i < 5; i++ {
```
Comment (mat007) [inline pkg/e2e/compose_build_test.go:70]: I don’t think we need a loop here, it’s a deterministic bug.

File: pkg/e2e/compose_build_test.go [hunk 3]
```diff
@@ -66,6 +66,21 @@ func TestLocalComposeBuild(t *testing.T) {
 		res.Assert(t, icmd.Expected{Out: `"FOO": "BAR"`})
 	})
 
+	t.Run("build with multiple build-args ", func(t *testing.T) {
```
Comment (mat007) [inline pkg/e2e/compose_build_test.go:69]:
```suggestion
	t.Run("build with multiple build-args", func(t *testing.T) {
```

File: pkg/e2e/fixtures/build-test/multi-args/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ARG IMAGE=666
+ARG TAG=666
+
+FROM ${IMAGE}:${TAG}
```
Comment (mat007) [inline pkg/e2e/fixtures/build-test/multi-args/Dockerfile:4]: We need a way to input two different values for the args, and afterwards verify that they’re still two different values.

File: pkg/e2e/compose_build_test.go (hunk 2 above)
Comment (ulyssessouza) [reply pkg/e2e/compose_build_test.go:70]: That was just a desperate attempt to reproduce it 🥲

File: pkg/e2e/compose_build_test.go [hunk 5]
```diff
@@ -66,6 +66,21 @@ func TestLocalComposeBuild(t *testing.T) {
...
+		for i := 0; i < 5; i++ {
+			// ensure local test run does not reuse previously build image
+			c.RunDockerOrExitError("rmi", "-f", "multiargs")
```
Comment (mat007) [inline pkg/e2e/compose_build_test.go:72]: Your test works, it’s just that here it’s
```suggestion
		c.RunDockerOrExitError("rmi", "-f", "multi-args_multiargs")
```

**Comment by mat007** [comment]
I updated the PR.

**Review by mat007 — APPROVED** [review]

**Pull request merged**